File: Examples/Misc/C#/Examples/MiscExamples/Examples/Example03.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;
using ExampleBase;

using Outlook = NetOffice.OutlookApi;

namespace MiscExamplesCS4
{
    /*
        *  in some situations you want check for a running office application instance.
        *  this example shows you how to use the Marshal.GetActiveObject method to get a running application and create a NetOffice wrapper instance.
        *  for this example we use outlook. please note the Marshal.GetActiveObject method throws a COMException if no running instance available
    */
    public partial class Example03 : UserControl, IExample
    {
        IHost _hostApplication;

        public Example03()
        {
            InitializeComponent();
        }
         
        #region IExample Member

        public void RunExample()
        {
            // its an example with an own visual control
            // checkout buttonStartExample_Click
        }

        public void Connect(IHost hostApplication)
        {
            _hostApplication = hostApplication;
        }

        public string Caption
        {
            get { return _hostApplication.LCID == 1033 ? "Example03" : "Beispiel03"; }
        }

        public string Description
        {
            get { return _hostApplication.LCID == 1033 ? "How to access a running Outlook application" : "Eine laufene Outlook Instanz automatisieren"; }
        }

        public UserControl Panel
        {
            get { return this; }
        }

        #endregion

        #region UI Trigger

        private void buttonStartExample_Click(object sender, EventArgs e)
        {
            // Initialize NetOffice
            LateBindingApi.Core.Factory.Initialize();

            try
            {
                Outlook.Application application = null;

                object nativeProxy = System.Runtime.InteropServices.Marshal.GetActiveObject("Outlook.Application");
                application = new Outlook.Application(null, nativeProxy);

                textBoxLog.Clear();
                textBoxLog.AppendText("we got running outlook instance\r\n");
                textBoxLog.AppendText("outlook version is " + application.Version);

                // instance was already running at start. we dispose references but not quit application
                application.Dispose();
            }
            catch (System.Runtime.InteropServices.COMException exception)
            {
                _hostApplication.ShowErrorDialog(null, exception);
            }            
        }
        
        #endregion
    }
}

File: Examples/Misc/VB/Examples/MiscExamples/Examples/Example03.vb
Imports ExampleBase
Imports LateBindingApi.Core
Imports Outlook = NetOffice.OutlookApi

'  in some situations you want check for a running office application instance.
'  this example shows you how to use the Marshal.GetActiveObject method to get a running application and create a NetOffice wrapper instance.
'  for this example we use outlook. please note the Marshal.GetActiveObject method throws a COMException if no running instance available
Public Class Example03
    Implements IExample

    Dim _hostApplication As ExampleBase.IHost
    
    Public Sub New()

        InitializeComponent()
 
    End Sub

#Region "IExample Member"

    Public Sub RunExample() Implements ExampleBase.IExample.RunExample

        ' its an example with an own visual control
        ' checkout buttonStartExample_Click

    End Sub

    Public ReadOnly Property Caption As String Implements ExampleBase.IExample.Caption
        Get
            Return IIf(_hostApplication.LCID = 1033, "Example03", "Beispiel03")
        End Get
    End Property

    Public ReadOnly Property Description As String Implements ExampleBase.IExample.Description
        Get
            Return IIf(_hostApplication.LCID = 1033, "How to access a running Outlook application", "Eine laufene Outlook Instanz automatisieren")
        End Get
    End Property

    Public Sub Connect(ByVal hostApplication As ExampleBase.IHost) Implements ExampleBase.IExample.Connect

        _hostApplication = hostApplication

    End Sub

    Public ReadOnly Property Panel As System.Windows.Forms.UserControl Implements ExampleBase.IExample.Panel
        Get
            Return Me
        End Get
    End Property

#End Region

#Region "UI Trigger"

    Private Sub buttonStartExample_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles buttonStartExample.Click

        ' Initialize NetOffice
        LateBindingApi.Core.Factory.Initialize()

        Try

            Dim application As Outlook.Application = Nothing
            Dim nativeProxy As Object = System.Runtime.InteropServices.Marshal.GetActiveObject("Outlook.Application")

            application = New Outlook.Application(Nothing, nativeProxy)
            textBoxLog.Clear()
            textBoxLog.AppendText("we got running outlook instance" + vbNewLine)
            textBoxLog.AppendText("outlook version is " + application.Version)

            'instance was already running at start. we dispose references but not quit application
            application.Dispose()

        Catch ex As System.Runtime.InteropServices.COMException

            _hostApplication.ShowErrorDialog(Nothing, ex)

        End Try

    End Sub

#End Region

End Class
